runners/core-java/src/test/java/org/apache/beam/runners/core/SimpleDoFnRunnerTest.java

```diff
@@ -386,6 +399,180 @@ public void testInfiniteSkew() {
             BoundedWindow.TIMESTAMP_MAX_VALUE));
   }
 
+  @Test
+  @Category({ValidatesRunner.class})
+  public void testRunnerNoSkew() {
+    PCollection<Duration> input =
+        p.apply("create", Create.timestamped(Arrays.asList(new Duration(0L), new Duration(1L)), Arrays.asList(0L, 1L)));
+    input.apply(ParDo.of(new SkewingDoFn(Duration.ZERO)));
+    // The errors differ between runners but at least check that the output timestamp is printed.
+    thrown.expectMessage(String.format("%s", new Instant(0L)));
```

Comment:
Note that you can solve this problem by having the DoFn catch the exception from within the processElement method outputting `success` or `exception stack trace/message` messages and performing a PAssert on the output PCollection.

Reply:
That's a neat idea. though it still can't make any extra validations on the error message thrown from the system though and would be limited to the same check. I guess we could check more specifically where it failed, but not sure of the overall value there. Though if left as is it keeps the error checking in the test code instead of the DoFn which I like. So I'll probably leave it as is unless you have a strong opinion.